impl/src/main/java/org/pentaho/di/trans/dataservice/optimization/ValueMetaResolver.java

```diff
@@ -66,15 +66,21 @@ public ValueMetaResolver( RowMetaInterface rowMeta ) {
 
         switch ( valueMetaInterface.getType() ) {
           case ValueMetaInterface.TYPE_DATE:
-            valueMetaInterface.setConversionMask( ANSI_DATE_LITERAL );
+            if ( valueMetaInterface.getConversionMask() == null ) {
```

Comment:
Shouldn't we check for null or empty string?

Reply:
@joaormf Yes, agree with that, if the mask is empty we will fall in the same error. I will update this.